sdk/storage/storage-blob/test/blobclient.spec.ts

```diff
@@ -483,6 +483,26 @@ describe("BlobClient", () => {
     }
   });
 
+  it("setAccessTier with snapshot", async () => {
+    const resp = await blockBlobClient.createSnapshot();
+    const blockBlobClientWithSnapshot = blockBlobClient.withSnapshot(resp.snapshot!);
+
+    await blockBlobClientWithSnapshot.setAccessTier("Cool");
+
+    const properties = await blockBlobClientWithSnapshot.getProperties();
+    assert.equal(properties.accessTier!.toLowerCase(), "cool");
```

Comment:
minor: maybe assert on accessTier before using `!`?
```suggestion
    assert.ok(properties.accessTier);
    assert.equal(properties.accessTier!.toLowerCase(), "cool");
```